gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/CodeableConceptMapper.java

```diff
@@ -196,6 +204,31 @@ private CodeableConcept generateCodeableConceptUsingConceptId(CD mainCode, CD tr
         return createCodeableConcept(conceptId, SNOMED_SYSTEM, display, text, extension);
     }
 
+    private void addNonSnomedCodesToCodeableConcept(CodeableConcept codeableConcept, CD codedData) {
+        if (codedData.hasCodeSystem() && !SNOMED_SYSTEM_CODE.equals(codedData.getCodeSystem())) {
+            var coding = mapNonSnomedCodes(codedData);
+            codeableConcept.getCoding().add(coding);
+        }
+
+        var translationCodes = codedData
+                .getTranslation()
+                .stream()
+                .filter(cd -> !SNOMED_SYSTEM_CODE.equals(cd.getCodeSystem()))
+                .map(this::mapNonSnomedCodes)
+                .toList();
+
+        if (!translationCodes.isEmpty()) {
+            codeableConcept.getCoding().addAll(translationCodes);
+        }
+    }
+
+    private Coding mapNonSnomedCodes(CD codedData) {
+        return new Coding()
+                .setCode(codedData.getCode())
+                .setSystem(CodeSystemsUtil.getFhirCodeSystem(codedData.getCodeSystem()))
```

Comment:
`CodeSystemsUtil.getFhirCodeSystem` appears to return the parameter if it isn't aware of the system. Is that behaviour fine for this, or should we be doing something else?

Reply:
Yes thats correct, if we don't know the FHIR code system, then we should return the HL7 code system, discussed with Emile in Slack

`Thank you , just read the doc - looks exactly right. The only question is what happens if you cannot translate the HL7 XML Code System (2.16.840.1.113883.2.1.6.3) to a FHIR code system (e.g. https://fhir.hl7.org.uk/Id/egton-codes)? Would you just output the raw value like 2.16.840.1.113883.2.1.6.3`


`
Yes that is correct, if we cannot map a code known code system, then we will output the Raw value. 
`

Reply:
Fair enough. I don't have an opinion on what the business requirements should be, just felt a bit strange at first.